cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+	}
+	return ""
+
+}
+
+// This function is parsing a grub.cfg file
+// input: absolute directory path to grub.cfg
+// output: Return a list of strings with the following format
+//	 line[3*x] - menuconfig
+//	 line[3*x+1] - linux kernel + boot options
+// 	 line[3*x+2] - initrd
+// and the default boot entry configured into grub.cfg
+
+func get_file_menu_content(path string) ([]string, int) {
+	var return_value []string
+	file, _ := os.Open(path + "/grub.cfg")
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+	}
+	var status int
+	var int_return int
+	int_return = 0
+	status = 0
+	// When status = 0 we are looking for a menu entry
+	// When status = 1 we are looking for a linux entry
+	// When status = 2 we are looking for a initrd entry
+	var trimmed_line string
+	for scanner.Scan() {
+		trimmed_line = strings.TrimSpace(scanner.Text())
+		trimmed_line = strings.Join(strings.Fields(trimmed_line), " ")
+		if strings.HasPrefix(trimmed_line, "#") == false {
+			if (strings.HasPrefix(trimmed_line, "set default=") == true) && (status == 0) {
+				fmt.Sscanf(trimmed_line, "set default=\"%d\"", &int_return)
+			}
+			if (strings.HasPrefix(trimmed_line, "menuentry ") == true) && (status == 0) {
+				status = 1
+				return_value = append(return_value, trimmed_line)
+			}
+			if (strings.HasPrefix(trimmed_line, "linux ") == true) && (status == 1) {
+				status = 2
+				return_value = append(return_value, trimmed_line)
+			}
+			if (strings.HasPrefix(trimmed_line, "initrd ") == true) && (status == 2) {
+				status = 0
+				return_value = append(return_value, trimmed_line)
+			}
+		}
+	}
+	return return_value, int_return
+
+}
+
+func copy_local(path string) string {
+	var dest string
+	result := strings.Split(path, "/")
+	for _, entry := range result {
+		dest = entry
+	}
+	dest = "/tmp/" + dest
+	srcFile, err := os.Open(path)
+	check(err)
+	defer srcFile.Close()
+
+	destFile, err := os.Create(dest) // creates if file doesn't exist
+	check(err)
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, srcFile) // check first var for number of bytes copied
+	check(err)
+
+	err = destFile.Sync()
+	check(err)
+	return dest
+}
+
+// This function is booting new kernel based on the content of grub.cfg
+
+func kexec_entry(grub_conf_path string, mount_point string) {
+	var file_menu_content []string
+	var entry int
+	var local_kernel_path string
+	var local_initrd_path string
+	if verbose {
+		println(grub_conf_path)
+	}
+	file_menu_content, entry = get_file_menu_content(grub_conf_path)
+	var kernel string
+	var kernel_parameter string
+	var initrd string
+	var kernel_infos []string
+	kernel_infos = strings.Fields(file_menu_content[3*entry+1])
+	kernel = kernel_infos[1]
+	var count int
+	count = 0
+	for _, field := range kernel_infos {
+		if count > 1 {
+			kernel_parameter = kernel_parameter + " " + field
+		}
+		count = count + 1
+	}
+	fmt.Sscanf(file_menu_content[3*entry+2], "initrd %s", &initrd)
```

Comment:
what if there is no separate initrd?

kexec.FileLoad can take ramfs = nil in this case.